### PR TITLE
Fix some push issues by adjusting heartbeat interval

### PR DESCRIPTION
**Pull request by grote**
When visiting settings UI and one is not checked in, this will also cause a check-in to be performed (if enabled). In case internet is missing when check-in is attempted, it will be retried once internet is back.

### CODE REVIEW

File: play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt [hunk 1]
```diff
@@ -56,24 +60,30 @@ class SettingsFragment : ResourceSettingsFragment() {
...
 
     private suspend fun updateDetails() {
-        if (getGcmServiceInfo(requireContext()).configuration.enabled) {
+        val context = requireContext()
+        val gcmServiceInfo = getGcmServiceInfo(context)
+        if (gcmServiceInfo.configuration.enabled) {
             val database = GcmDatabase(context)
             val regCount = database.registrationList.size
+            // check if we are connected as we should be and re-connect if not
+            if (!McsService.isConnected(context)) {
```
Comment (grote) [inline play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt:69]: I think `McsService` is in the persistent process, so we can't/shouldn't do this check here like this, right?

Comment (grote) [reply play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt]: I changed this to use the `gcmServiceInfo` directly.

**Comment by chirayudesai** [comment]
I've been running this for a few months now, and it's working well.

Push notifications would be delayed sometimes before but now they're pretty much instantaneous.

**Comment by eggplanting** [comment]
> I've been running this for a few months now, and it's working well.
> 
> Push notifications would be delayed sometimes before but now they're pretty much instantaneous.

Tested your updated APK posted here:
https://www.reddit.com/r/CalyxOS/comments/pkvx81/microg_push_notification_fixes_test_apks/

Only with them push notifications seem to work first time to me, without any issues.

I guess many people would appreciate if the fixes will be integrated in the next stable release.

Thank you all for making this possible!

**Review by emkangail — APPROVED** [review]

File: play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt (hunk 1 above)
Comment (emkangail) [reply play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt]: Thanks

**Comment by ale5000-git** [comment]
@grote: The CI now report a compilation error.

**Comment by grote** [comment]
Looks like @mar-v-in merged master in here. I can rebase this on master and resolve issues.

**Comment by mar-v-in** [comment]
@grote that would be amazing :)

**Pull request merged**